developerTools/scriptReferences/entityClientScript/entityClientScript.js

```diff
@@ -0,0 +1,116 @@
+//
+// entityClientScript.js
+//
+// Adapted from script https://github.com/robin-k-wilson/hifi-dev-utils/blob/master/client_entity_all_events.js
+// Updated by Robin Wilson 4/15/2019
+//
+// Generic client entity methods and signals that are triggered by mouse or controller interactions
+// Use as a template for your client entity scripts
+//
+// Client entity scripts are run on a interface instance. Meaning each avatar connected
+// to a domain have one instance of this script running locally and one client controls
+// an entity at one time called "ownership over that entity".
+// Best for physics changes.
+//
+// Copyright 2019 High Fidelity, Inc.
+//
+// Distributed under the Apache License, Version 2.0.
+// See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+(function() {
+    var I_AM_STATIC_VARIABLE = 200;
+    var _this = null;
+    var _entityID = null;
+
+    function EntityClientScript() {
+        _this = this;
+    }
+
+    EntityClientScript.prototype = {
+        // REQUIRED FUNCTIONS
+        preload: function(entityID) {
+            // runs when this script is refreshed or a
+            // client connects to a domain where this entity is present
+            _entityID = entityID;
```

Comment:
can you please make `_entityID` a member of `_this` instead of its own variable? if you don't want to, please help me understand why not :)